web/src/beta/features/Editor/Settings/hooks.ts

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect, useMemo } from "react";
+
+import { NLSLayer } from "@reearth/services/api/layersApi/utils";
+import { Page } from "@reearth/services/api/storytellingApi/utils";
+
+type SettingProps = {
+  onPageUpdate?: (id: string, layers: string[]) => void;
+  layers?: NLSLayer[];
+  selectedPage?: Page;
+};
+export default ({ layers, selectedPage, onPageUpdate }: SettingProps) => {
+  const pageId = selectedPage?.id;
+
+  const allLayerIds = useMemo(() => {
+    return layers?.map(layer => layer.id) || [];
+  }, [layers]);
```

Comment:
Since this is only used in allLayersSelected, please just put it there in that useMemo to keep code a bit more tidy.